Automattic/jetpack
# VideoPress: replace the usage of useConnection() by local helper function

The `@jetpack-connetion` pkg doesn't support the site status for Simple sites. For this, this PR removes its usage and relies on the `JP_CONNECTION_INITIAL_STATE` and `videoPressEditorState` global vars to know the user's connection status.

This is performed by the new `isUserConnected()` helper function. The function debug()s the status.

Fixes #

## Proposed changes:

* VideoPress: replace the usage of useContext() by local helper function

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the Block editor
* Enable and filter the debug() lines
```
localStorage.setItem( 'debug', 'videopress:*' )
```
* Hard refresh
* Add a VideoPress video block
* Confirm the logs are consistent with the connection status. Connect/disconnect Jetpack to check it.

<img width="507" alt="image" src="https://user-images.githubusercontent.com/77539/215082827-788c256c-c070-4d8b-a21e-b0eef43f1cc5.png">

<img width="591" alt="image" src="https://user-images.githubusercontent.com/77539/215083509-450728ef-5f4e-4839-9ce9-2df8799afc75.png">
